File: live-test/java/src/test/java/demo/ProbeTest.java
// Scope-discrimination fixture, test side: hosts a test-source caller and a
// test-source subtype of the production probe class, so test-scoped results
// differ from production-scoped ones.
package demo;

class ProbeTest {
    void testCaller() {
        new Probe().target();
    }
}

// Test-side subtype: listed only when hierarchy scope includes test sources.
class ProbeTestChild extends Probe {
}

File: live-test/kotlin/src/test/kotlin/demo/ProbeTest.kt
// Test-source side of the scope-discrimination pair: testCaller and
// ProbeTestChild give the production probe class a caller and a subtype that
// only test-inclusive search scopes should surface.
package demo

class ProbeTest {
    fun testCaller() {
        Probe().target()
    }
}

class ProbeTestChild : Probe()
